game-core/src/main/java/games/strategy/triplea/delegate/DiceRoll.java

```diff
@@ -803,7 +797,7 @@ public static DiceRoll rollNDice(
                 UnitSupportAttachment::getStrength);
       } else {
         strength = ua.getAttack(unit.getOwner());
-        if (ua.getIsMarine() != 0 && isAmphibiousBattle && amphibiousLandAttackers.contains(unit)) {
```

Comment:
Have you tested if marines are being given an amphib assault bonus on standard land battles? I'd double check that looking at this update. Have you run across where 'getWasAmphibious' is set to true (could you point me there)?

Reply:
`getWasAmphibious` is only set in `TransportTracker#unloadTransportChange` and that is when a transport is unloaded.  I just ran a game where I did an amphibious assault and then in the next turn used the same units to attack a standard land battle.  `wasAmphibious` was true for the first battle but false for the second battle.  I've also tested battles that don't have any amphibious assault and `wasAmphibious` is false there as well.

Reply:
Interesting, that implies/confirms at one point a new way to track amphibious units was simply built without regard to the existing mechanism. This is then a very good example of SRP / DRY being violated. 

It might be worthwhile to walk through this code execution in a debugger to make sure it all looks good at that level. It sounds like the test cases you've ran through have been pretty comprehensive.

Reply:
Yeah, I went through it with a debugger to make sure the value was what I expected.  If you have two units attacking and one was amphibious but the other goes by land, only the amphibious unit has `wasAmphibious`.

Looking at the history, it appears that `amphibiousLandAttackers` was first created to track if a battle was amphibious.  Then, when marine bonuses were added, the `wasAmphibious` was added but it was used oddly (it kind of determined if a battle was amphibious but wasn't used to check if the unit should get a marine bonus).  Then that code had concurrent issues because it talked to the battle tracker.  It was finally removed and replaced by plumbing `amphibiousLandAttackers` through.

I think this new way (using `wasAmphibious`) should work correctly and it should be simpler.

Reply:
:+1: